lib/kubernetes-deploy/cluster_resource_discovery.rb

```diff
@@ -16,8 +16,24 @@ def crds
       end
     end
 
+    def globals
+      @globals ||= fetch_globals.map do |gv|
+        kind, group = gv.split(".", 2)
+        { group: group, kind: kind.singularize }
```

Comment:
`singularize` isn't necessarily going to give us the real kind. You can get it from kubectl if you use `-o wide` instead of `-o name`. Seems like the kind is the only part you actually use anyway, so maybe this should just be `global_kinds` and return an array?

Incidentally, I noticed the docs have a worrying disclaimer: "(This output can be retrieved from kubectl api-resources, and is accurate as of Kubernetes 1.13.3.)" but it [turns out](https://github.com/kubernetes/website/pull/13223) that was supposed to mean that the copy of the output mentioned in the docs was snapshotted from 1.13.3. So this should be fine to use, as you'd hope.

Reply:
I'd prefer to use `singularize` over parsing the output to wide, but 👍 to the array approach.

Reply:
It really isn't safe to assume that `singularize` (some logic somebody wrote into ruby) will generate the real `kind` name (which is explicitly chosen by someone writing a new resource). In theory the plural, the singular and the kind could all be completely different words. I'm not sure there are examples of unexpected plural forms in core resources, but CRs can be global too.

Reply:
Good point, code now parses wide. This was a bit tricker than expected, some columns don't have values so just splitting on whitespace doesn't work, instead I took advantage of the fixed width of each column.